### PR TITLE
authentication

### PR DESCRIPTION
**Pull request by banukutlu**
part of issue #7 
also needs specs #19

### CODE REVIEW

**Comment by cdmo** [comment]
yes, needs specs: https://github.com/psu-libraries/myaccount/issues/19

**Comment by cdmo** [comment]
What do you think about updating the README.md with some descriptions of some of this stuff, particularly that we are using Warden and how it works with `config` and what you have to do to make that work locally? Other stuff?

File: Gemfile [hunk 1]
```diff
@@ -6,10 +6,13 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 ruby '2.6.5'
 
 gem 'bootsnap', '>= 1.4.2', require: false
+gem 'config'
+gem 'http'
```
Comment (cdmo) [inline Gemfile]:
```suggestion
gem 'http', '~> 4.0'
```

Let's peg to a version. That ok?

Comment (cdmo) [reply Gemfile]: Same conversation for `warden` and `config` too of course

**Comment by cdmo** [comment]
What are the changes to the yml files? Like `config/settings/development.yml`? Just empty files... 🤔

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# frozen_string_literal: true
+
+class SessionsController < ApplicationController
+  def index
+    # @symphony_ok = symphony_client.ping
```
Comment (cdmo) [inline app/controllers/sessions_controller.rb]:
```suggestion
```

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+    redirect_to summaries_url if current_user? || request.env['warden'].authenticate(:library_id)
+    # TODO: redirect_to errors_url, alert: 'Unable to authenticate.'
```
Comment (cdmo) [inline app/controllers/sessions_controller.rb]: see #20

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+    # TODO: flash.now.alert = request.env['warden'].message
```
Comment (cdmo) [inline app/controllers/sessions_controller.rb]: What is there todo for this one?

**Comment by banukutlu** [comment]
> What are the changes to the yml files? Like `config/settings/development.yml`? Just empty files... 🤔

yes they are empty so far, got added as part of config:install

File: app/controllers/sessions_controller.rb (hunk 4 above)
Comment (banukutlu) [reply app/controllers/sessions_controller.rb]: for the case if login was not successful, display the message kinda thing 🤔

Comment (cdmo) [reply app/controllers/sessions_controller.rb]: So we can delete, because I created #20 ?

Comment (banukutlu) [reply app/controllers/sessions_controller.rb]: sure, will do

**Comment by banukutlu** [comment]
> Requesting a README update (I guess this is part of #11)

updated 👍

**Pull request merged**